tasks/task04/task04.py

```diff
@@ -0,0 +1,27 @@
+def collect_leaves(i) -> list:
+    res = []
+    if type(i) is list:
+        return i
+    for node, values in i.items():
```

Comment:
here I would add check if `i` as a dict, so you are sure there will be `items` available

Reply:
check added